src/useAsyncRetry.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback, useState } from 'react';
+import useAsync from './useAsync';
+
+const useAsyncRetry = <T>(fn: () => Promise<T>, args: any[] = []) => {
+  const [attempt, setAttempt] = useState<number>(0);
+  const memoized = useCallback(async () => await fn(), [...args, attempt]);
+  const state = useAsync(memoized);
+
+  const retry = useCallback(() => {
+    if (state.loading) {
+      return;
```

Comment:
Shall we warn user in development mode that she is trying to "retry" while promise is being resolved?

```js
if (state.loading) {
  if (process.env.NODE_ENV !== 'production') {
    console.warn('You are calling useAsyncRetry hook retry() method while loading in progress, this is a no-op.');
  }
  return;
}
``

Reply:
Yep, I had a console message originally for debugging purposes. I think I'm more inclined to do a `process.env.NODE_ENV === "development"` or to use `console.log` to avoid aborting build pipelines (CRA aborts pipeline when `CI=true` and any `console.warn` or `console.error` messages are outputted).